packages/react-aria-components/src/Link.tsx

```diff
@@ -55,7 +55,7 @@ export const LinkContext = createContext<ContextValue<LinkProps, HTMLAnchorEleme
 function Link(props: LinkProps, ref: ForwardedRef<HTMLAnchorElement>) {
   [props, ref] = useContextProps(props, ref, LinkContext);
 
-  let elementType = typeof props.children === 'string' ? 'span' : 'a';
+  let elementType = typeof props.children === 'string' || typeof props.children === 'function' ? 'span' : 'a';
```

Comment:
this one is going to be a little funky since we can't see what is being rendered by a function child, I'll bring it to the team